go/teams/chain.go

```diff
@@ -1602,9 +1645,35 @@ func (t *teamSigchainPlayer) addInnerLink(
 			return res, fmt.Errorf("unsupported link type: %s", payload.Body.Type)
 		}
 	}
-	if isHighLink {
-		res.newState.inner.LastHighLinkID = link.LinkID().Export()
-		res.newState.inner.LastHighSeqno = link.Seqno()
+
+	// We only run addInnerLink out of order when inflating, and so we skip
+	// computing high skips during it.
+	// updates high skip data, so we skip it. If highSkip is presented and
+	// either we don't need to reverify or we want to check it, check it.
+	highSkip := payload.HighSkip
+	if highSkipVerifyDesired {
+		if highSkip != nil && !reverifyNeeded {
```

Comment:
I think `&& !reverifyNeeded` is redundant here. May want to remove it or if you're feeling real cautious error out instead of skipping verification.